gestalt/provider/vault.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional, Tuple, Any
+import hvac  # type: ignore
+import os
+
+
+class Vault(Provider):
+    def __init__(self,
+                 cert: Optional[Tuple[str, str]] = None,
+                 role: Optional[str] = None,
+                 jwt: Optional[str] = None,
+                 url: Optional[str] = os.environ.get("VAULT_ADDR"),
+                 token: Optional[str] = os.environ.get("VAULT_TOKEN"),
+                 verify: Optional[bool] = True) -> None:
+        """Initialized vault client and authenticates vault
+
+        Args:
+            client_config (HVAC_ClientConfig): initializes vault. URL can be set in VAULT_ADDR
+                environment variable, token can be set to VAULT_TOKEN environment variable.
+                These will be picked by default if not set to empty string
+            auth_config (HVAC_ClientAuthentication): authenticates the initialized vault client
+                with role and jwt string from kubernetes
+        """
+        self.vault_client = hvac.Client(url=url,
+                                        token=token,
+                                        cert=cert,
+                                        verify=verify)
+        try:
+            self.vault_client.is_authenticated()
+        except requests.exceptions.MissingSchema:
+            raise RuntimeError(
+                "Gestalt Error: Unable to connect to vault with the given configuration"
+            )
+        if role and jwt:
+            try:
+                self.vault_client.auth_kubernetes(role=role, jwt=jwt)
+            except hvac.exceptions.InvalidPath:
+                raise RuntimeError(
+                    "Gestalt Error: Kubernetes auth couldn't be performed")
+            except requests.exceptions.ConnectionError:
+                raise RuntimeError("Gestalt Error: Couldn't connect to Vault")
+
+    def get(self, key: str, path: str, filter: str) -> Any:
+        """Gets secret from vault
+        Args:
+            key (str): key to get secret from
+            path (str): path to secret
+            filter (str): filter to apply to secret
+        Returns:
+            secret (str): secret
+        """
+        try:
+            response = self.vault_client.read(path)
+            if response is None:
+                raise RuntimeError("Gestalt Error: No secrets found")
+            requested_data = response['data']['data']
+        except hvac.exceptions.InvalidPath:
+            raise RuntimeError(
+                "Gestalt Error: The secret path or mount is set incorrectly")
+        except requests.exceptions.ConnectionError:
+            raise RuntimeError(
+                "Gestalt Error: Gestalt couldn't connect to Vault")
+        except Exception as err:
+            raise RuntimeError(f"Gestalt Error: {err}")
+        if filter is None:
+            return requested_data
+        filter = "." + filter
+        secret = requested_data
```

Comment:
i planned to leave the `.` as part of user input. Typically when I see user provided json pathing it's expected. dollar sign not so much.

Reply:
sounds good. Updating in the next push